### PR TITLE
Update the README and high level documentation [ready]

### PR DESCRIPTION
**Pull request by amiller**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,21 +3,88 @@
...
+Other MPC toolkits, such as SCALE-MAMBA, Viff, EMP, SPDZ, and others, do not provide guaranteed output delivery, and so if even a single node crashes they stop providing output at all.
+
+Linear communication overhead is about scaling to large network sizes. HoneyBadgerMPC implements aggressive batching and amortization, so as more server nodes are added, the communication cost per server approaches a constant). 
+
+### Comparison with zero knowledge proofs
+As a blockchain privacy layer, MPC is seen as an alternative to commitments and zero knowledge proofs, which are already widely used in cryptocurrencies such as Zcash, Monero, Ethereum, etc.
+The main difference is that with zero knowledge proofs, typically only a single party (the prover) knows the opening of the committed values (the witness), so if this party crashes, the data is no longer available.
+In MPC, the witness is secret shared, so it is guaranteed to be available as long as `n-t` of the servers are running correctly.  the secret data is guaranteed to be available for subsequent computations, where as with zero knowledge proofs
```
Comment (sbellem) [inline README.md]: >  correctly.  the secret
The ...

File: README.md [hunk 2]
```diff
@@ -3,21 +3,88 @@
...
+
+Linear communication overhead is about scaling to large network sizes. HoneyBadgerMPC implements aggressive batching and amortization, so as more server nodes are added, the communication cost per server approaches a constant). 
+
+### Comparison with zero knowledge proofs
+As a blockchain privacy layer, MPC is seen as an alternative to commitments and zero knowledge proofs, which are already widely used in cryptocurrencies such as Zcash, Monero, Ethereum, etc.
+The main difference is that with zero knowledge proofs, typically only a single party (the prover) knows the opening of the committed values (the witness), so if this party crashes, the data is no longer available.
+In MPC, the witness is secret shared, so it is guaranteed to be available as long as `n-t` of the servers are running correctly.  the secret data is guaranteed to be available for subsequent computations, where as with zero knowledge proofs 
+blockchain privacy techniques, such as Zcash and Moner oThe main difference between MPC and  that the secret data is always available, without having any individual knowing the secret.
```
Comment (sbellem) [inline README.md]: > and Moner oThe 
and Monero. The ...

File: README.md [hunk 3]
```diff
@@ -3,21 +3,88 @@
...
+, and can provide availability where zero knowledge proofs get stuck
+So far, blockchain privacy has made use of Zero Knowledge proofs.
```
Comment (sbellem) [inline README.md]: It looks like there was an edit "mixup" after the sentence

> In MPC, the witness is secret shared, so it is guaranteed to be available as long as `n-t` of the servers are running correctly.

not sure what the exact intended phrasing is ... maybe something like:

"The main difference between MPC and blockchain privacy techniques, such as Zcash and Monero is that the secret data is always available, without having any individual knowing the secret, and can provide availability where zero knowledge proofs get stuck. So far, blockchain privacy has made use of Zero Knowledge proofs."

File: README.md [hunk 4]
```diff
@@ -3,21 +3,88 @@
...
+The programming model is based on Python mixins, which extend a `Share` that represents a pipelined MPC computation.
 
-## [HoneyBadgerMPC Subprotocols](docs/subprotocols.rst)
-See [docs/subprotocols.rst](docs/subprotocols.rst).
+To reach agreement on the published inputs, we make use of either the built-in asynchronous broadcast protocol (a port of HoneyBadgerBFT), or else an external blockchain service (See Fabric Integration and Ethereum Integration).
+
+2. Online phase.
+   Once clients provide input, the protocol computes an MPC program layer by layer. 
+   An MPC program consist of linear operations on secret shared data, as well as batch reconstruction. Linear operations can be computed locally, non-linear operations like multiplication must be emulated using share reconstruction and preprocessed values.
+   See `mpc.py`, `batch_reconstruction.py`, `taskprogramrunner.py`, `reed_solomon.py`
```
Comment (Drake-Eidukas) [inline README.md:32]: We should make these actual links to the files

**Review by Drake-Eidukas — APPROVED** [review]
Small nit, otherwise looks good!

File: README.md (hunk 4 above)
Comment (amiller) [reply README.md:32]: Could you take care of this ^

**Pull request merged**